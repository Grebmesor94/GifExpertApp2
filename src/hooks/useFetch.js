import { useState, useEffect } from "react"
import { getGifs } from "../helpers/getGifs"


export const useFetch = (category) => {
  const [images, setImages] = useState([])
  const [loading, setLoading] = useState(true)

  const getImages = async () => { 
    const newImages = await getGifs(category)
    setImages(newImages)
    setLoading(false)
  }

  useEffect(() => {
  
    getImages(category)
 
  }, [category])
 
  return {
    images,
    loading
  }
  
}
