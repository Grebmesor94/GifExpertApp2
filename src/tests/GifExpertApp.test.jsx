import { fireEvent, render, renderHook, screen, waitFor } from "@testing-library/react";
import { GifExpertApp } from "../GifExpertApp";
import { useFetch } from "../hooks/useFetch";


describe('test on <GifExpertApp />', () => {
  test('should match snapshot', () => {
    
    const {container} = render(<GifExpertApp />)

    expect(container).toMatchSnapshot();

  });
});