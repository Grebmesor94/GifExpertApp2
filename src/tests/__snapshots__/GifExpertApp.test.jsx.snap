// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifExpertApp /> should match snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="category"
      type="text"
      value=""
    />
  </form>
</div>
`;
